app/controllers/notifications_controller.rb

```diff
@@ -2,11 +2,37 @@ class NotificationsController < AuthenticatedController
   include ProjectScoped
 
   def index
-    @notifications = current_user.notifications.newest.limit(20)
-    @unread_count = @notifications.unread.count
-
+    # TODO this doesn't fix all the N+1 query problems, needs a closer look:
```

Comment:
What needs to be improved?

Reply:
If you look in the server output it looks like some N+1 queries are going on, I couldn't figure out an immediate fix so decided to leave it until later - but forgot to come back to it! Will try and fix now.